src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,15 @@
 package core.basesyntax;
 
 public class MainApp {
-
+    public static void main(String[] args) {
+        Machine truck = new Truck();
+        Machine bulldozer = new Bulldozer();
+        Machine excavator = new Excavator();
```

Comment:
The instances of `Machine` are correctly using abstract references. This is a good practice as it allows for polymorphism. However, make sure that the classes `Truck`, `Bulldozer`, and `Excavator` have informative `doWork` and `stopWork` method implementations that specify the type of machine that is working or stopping work.